### DISCUSSION TITLE
Please help - how to set input.focus() with react-hook-form?

### DISCUSSION THREAD

**Discussion by nicitaacom** [react-hook-form/react-hook-form#11216] [Q&A] [answered]
I want to set input focus state (when I click button and see dropdown with input)
i opened discussion where I tried to do it with useRef - it doesn't work for me
https://github.com/orgs/react-hook-form/discussions/11137
If you have minimal example - please share

If you don't understood what I want please let me know

**Comment by nicitaacom** [comment, marked as answer]
Second video on YouTube be query `react hook form auth focus`
https://youtu.be/WohMeMhLtwE?t=53

It's luck because this video not popular and if I close page with this YouTube query and open it again it show different videos

**Comment by nicitaacom** [comment]
```tsx
 useEffect(() => {
    //Timeout needed for focus and scroll to bottom - without it foucs and scrollToBottom doesn't work
    setTimeout(() => {
      inputRef.current?.focus()
      setFocus("message")
      if (bottomRef.current) {
        bottomRef.current.scrollTop = bottomRef.current.scrollHeight
      }
    }, 25)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isDropdown])
  ```

**Reply by allamgr** [reply]
Could you please explain why the Timeout is required? @nicitaacom

**Reply by nicitaacom** [reply]
I don't know - it so it work

Actually as you see I use inputRef and I forgot how I solved it (as I remember I used cruth to solve it)

So react-hook-form docs (in point of using focus) is bad because it doesn't work
